crates/storage/src/connection/event.rs

```diff
@@ -612,6 +600,21 @@ impl Transaction<'_> {
             .context("Querying bloom filter range")?
             .collect::<Result<Vec<_>, _>>()?;
 
+        // There are no aggregates in the database yet or the loaded aggregates
+        // don't cover the requested range.
+        let should_include_running = aggregates.last().map_or(true, |a| end_block > a.to_block);
+
+        if should_include_running {
+            let running_aggregate = match self.running_aggregate.lock() {
+                Ok(guard) => guard,
+                Err(poisoned) => {
+                    tracing::error!("Poisoned lock in load_aggregate_bloom_range");
+                    poisoned.into_inner()
+                }
+            };
```

Comment:
We don't have a uniform way of locking mutexes in the code and it's high time we had one. :thinking: